cmd/hyperkube/hyperkube.go

```diff
@@ -174,21 +176,41 @@ func (hk *HyperKube) Run(args []string) error {
 	logs.InitLogs()
 	defer logs.FlushLogs()
 
-	err = s.Run(s, s.Flags().Args())
-	if err != nil {
-		hk.Println("Error:", err)
+	if !s.RespectsStopCh {
+		// for command that do not respect the stopCh, we run them in a go
+		// routine while leaving them running while when stopCh is closed.
+		errCh := make(chan error)
+		go func() {
+			errCh <- s.Run(s, s.Flags().Args(), wait.NeverStop)
+		}()
+		select {
+		case <-stopCh:
+			return errors.New("interrupted") // this error text is ignored
+		case err := <-errCh:
+			if err != nil {
+				hk.Println("Error:", err)
+			}
+		}
+	} else {
+		err = s.Run(s, s.Flags().Args(), stopCh)
+		if err != nil {
+			hk.Println("Error:", err)
+		}
 	}
 
 	return err
 }
 
 // RunToExit will run the hyperkube and then call os.Exit with an appropriate exit code.
 func (hk *HyperKube) RunToExit(args []string) {
-	err := hk.Run(args)
-	if err != nil {
-		os.Exit(1)
+	stopCh, exitCodeCh := server.SetupSignalHandler()
+	exitCode := 1
+	defer func() { exitCodeCh <- exitCode }()
+
+	err := hk.Run(args, stopCh)
```

Comment:
Nit, but I like when these kind of things are written with the short-if. Ask @pweil-, he calls me short-if 👮